### ISSUE TITLE
Need welding system/component

### ISSUE THREAD

**Issue by ElectroJr** [space-wizards/space-station-14#6098]
Currently doors, entity storage (lockers, crates), and atmos scrubbers/vents all have welding related logic (Is-welded bools, visualizers, and do-after interactions). There should probably just be a single system that handles those interactions & sets welding appearance data.

**Comment by Willhelm53** [comment]
Is this still needed? Would it go under ToolSystem Welder or in its own system?

**Comment by mirrorcult** [comment]
> Is this still needed? Would it go under ToolSystem Welder or in its own system?

this is for things that can be welded, not welders themselves, so a different system

**Comment by mirrorcult** [comment]
a good analogue for this behavior would be how `LockComponent` and `LockSystem` work

**Comment by ElectroJr** [comment]
Yeah should have said "Weldable" system/component instead of "welding"

**Comment by mirrorcult** [comment]
this got done by macoron
